### PR TITLE
Update to ruby 2.4.4, Rails 5.1.6 + fix tests

### PR DESCRIPTION
**Pull request by nikolas**
The migration files needed a small compatibility change... see:

  https://github.com/RolifyCommunity/rolify/issues/444

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -13,39 +12,39 @@
...
-    t.datetime "updated_at"
-    t.integer  "branch_id"
-    t.string   "photo_file_name"
-    t.string   "photo_content_type"
-    t.integer  "photo_file_size"
+  create_table "leafs", force: :cascade do |t|
+    t.string "name"
+    t.text "content"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
Comment (mcfiredrill) [inline db/schema.rb]: This *shouldnt* break production, but if there are any records with NULL created_at/updated_at fields the migration will fail, I'm pretty sure.

Comment (mcfiredrill) [reply db/schema.rb]: The alternative is to mark all the timestamp fields `null: true` in the migration files. This change was sometime in the rails 5.x series I believe.

Comment (nikolas) [reply db/schema.rb]: Yeah..  i didn't do any of this, ... was just generated automatically for me. idk if it was the `rake db:migrate` or `rake setup`... kinda rusty with rails.

Comment (nikolas) [reply db/schema.rb]: that said i would be surprised if there are any Leafs on prod with an empty `created_at` or `updated_at`. I guess we can just check.

Comment (mcfiredrill) [reply db/schema.rb]: I think it probably happened when you ran `migrate` if you were starting on a fesh database. Its probably best to use `null: true` for the previous migrations. 

So we should add `null: true` option to all datetime fields in the old migrations.

Then we can create a new migration for changing the columns to `null: false` if we want to.

I mean, at least that's what I would do if this was a project at a company...maybe it doesn't make sense for an open source project.

**Comment by nikolas** [comment]
Alright, I've added `null: true` to all `t.timestamps` declarations in the migrations just to be safe. Let me know if that syntax is wrong or whatever.

File: Gemfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
 source 'https://rubygems.org'
 
-ruby "2.4.2"
```
[the remark this answers is not in the document]
Comment (nikolas) [reply Gemfile]: yea def... let's just merge this though since it's here and we can update to 2.5.1 once this is working.

**Comment by mcfiredrill** [comment]
> Alright, I've added null: true to all t.timestamps declarations in the migrations just to be safe. Let me know if that syntax is wrong or whatever.

This looks good, the problem now is the db/schema file still has them as `null: false`.  You can regenerate it if you run all the migrations again. Maybe you have to drop your development database though.

```
bundle exec rake db:drop
bundle exec rake db:migrate
```
Then commit the schema.rb file again.

**Comment by nikolas** [comment]
@mcfiredrill thanks for your help. I've done that, and squashed all these commits into a single one.

@houndci-bot thanks but you said that already.

**Comment by mcfiredrill** [comment]
Thanks! @nikolas

**Pull request merged**